packages/base/src/result.ts

```diff
@@ -0,0 +1,92 @@
+export enum ResultStatus {
+  Ok = 'Ok',
+  Error = 'Error',
+}
+export interface OkResult<TResult> {
+  status: ResultStatus.Ok
+  result: TResult
+}
+export interface ErrorResult<TError extends Error> {
+  status: ResultStatus.Error
+  error: TError
+}
+
+export type Result<TResult, TError extends Error> = OkResult<TResult> | ErrorResult<TError>
```

Comment:
we are going to write this ALL the time... maybe `Task` to save a few keystrokes :P

Reply:
Someone already defined a `Task` in `base` :P

Reply:
mmmm.... surely a really nice guy! :P